tests/integ/modin/binary/test_binary_op.py

```diff
@@ -1299,10 +1298,20 @@ def check_op(native_lhs, native_rhs, snow_lhs, snow_rhs):
             snow_ans, native_ans, lambda s: s, check_index_type=False
         )
 
-    check_op(lhs, rhs, try_cast_to_snow_series(lhs), try_cast_to_snow_series(rhs))
+    # The join count is high because:
+    # - When creating a single index Series, 1 join is performed; four series are created.
+    #   Therefore, 4 joins are performed. Each binary operation uses 1 join; two operations are performed.
+    #   This results in 6 joins.
+    # - Similarly, when creating a MultiIndex Series, 1 join is performed per column in the MultiIndex, in our case
+    #   there are two columns. Four Series are created, resulting in 8 joins. Each binary operation uses 1 join;
+    #   two operations are performed. This results in 10 joins.
+    with SqlCounter(
+        query_count=2, join_count=10 if isinstance(lhs.index, pd.MultiIndex) else 6
+    ):
+        check_op(lhs, rhs, try_cast_to_snow_series(lhs), try_cast_to_snow_series(rhs))
 
-    # commute series
-    check_op(rhs, lhs, try_cast_to_snow_series(rhs), try_cast_to_snow_series(lhs))
+        # commute series
+        check_op(rhs, lhs, try_cast_to_snow_series(rhs), try_cast_to_snow_series(lhs))
```

Comment:
why run it twice? let's use parameterize for switch lhs and rhs.

Reply:
done!